pkg/osquery/tables/gsettings/metadata.go

```diff
@@ -0,0 +1,274 @@
+	}
+
+	for _, schema := range schemas {
+		descriptions, err := t.gsettingsDescribeForSchema(ctx, schema)
+		if err != nil {
+			level.Info(t.logger).Log(
+				"msg", "error describing keys for schema",
+				"schema", schema,
+				"err", err,
+			)
+			continue
+		}
+		for _, d := range descriptions {
+			row := map[string]string{
+				"description": d.Description,
+				"type":        d.Type,
+				"schema":      schema,
+				"key":         d.Key,
+			}
+			results = append(results, row)
+		}
+
+	}
+
+	return results, nil
+}
+
+type datatype struct {
+	raw string
+}
+
+type keyDescription struct {
+	Description string
+	Type        string
+	Key         string
+}
+
+func (k *keyDescription) toMap() map[string]string {
+	return map[string]string{
+		"description": k.Description,
+		"type":        k.Type,
+		"key":         k.Key,
+	}
+}
+
+func (t *GsettingsMetadata) gsettingsDescribeForSchema(ctx context.Context, schema string) ([]keyDescription, error) {
+	var descriptions []keyDescription
+
+	keys, err := t.listKeys(ctx, schema)
+	if err != nil {
+		return descriptions, errors.Wrap(err, "fetching keys to describe")
+	}
+
+	for _, k := range keys {
+		desc, err := t.describeKey(ctx, schema, k)
+		if err != nil {
+			level.Info(t.logger).Log(
+				"msg", "error describing key",
+				"key", k,
+				"schema", schema,
+				"err", err,
+			)
+			continue
+		}
+		descriptions = append(descriptions, desc)
+	}
+
+	return descriptions, nil
+}
+
+func (t *GsettingsMetadata) listKeys(ctx context.Context, schema string) ([]string, error) {
+	var keys []string
+	output := new(bytes.Buffer)
+
+	err := t.cmdRunner(ctx, []string{"list-keys", schema}, output)
+	if err != nil {
+		return keys, errors.Wrap(err, "fetching keys")
+	}
+	scanner := bufio.NewScanner(output)
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		if line == "" {
+			continue
+		}
+
+		keys = append(keys, line)
+	}
+
+	if err := scanner.Err(); err != nil {
+		level.Info(t.logger).Log("msg", "scanner error", "err", err)
+	}
+
+	return keys, nil
+}
+
+func (t *GsettingsMetadata) describeKey(ctx context.Context, schema, key string) (keyDescription, error) {
+	desc := keyDescription{Key: key}
+
+	d, err := t.getDescription(ctx, schema, key)
+	if err != nil {
+		return desc, errors.Wrap(err, "getting key's description")
+	}
+	desc.Description = d
+
+	datatype, err := t.getType(ctx, key, schema)
+	if err != nil {
+		return desc, errors.Wrap(err, "discerning key's type")
+	}
+	desc.Type = datatype
+
+	return desc, nil
+}
+
+func (t *GsettingsMetadata) getDescription(ctx context.Context, schema, key string) (string, error) {
+	output := new(bytes.Buffer)
+
+	err := t.cmdRunner(ctx, []string{"describe", schema, key}, output)
+	if err != nil {
+		return "", errors.Wrap(err, "describing key")
+	}
+
+	return strings.TrimSpace(output.String()), nil
+}
+
+func (t *GsettingsMetadata) getType(ctx context.Context, key, schema string) (string, error) {
```

Comment:
The type conversion is weirdly spread across several things, and I'm not confident it's correct.  I think it's probably fine? I think it would also be fine to return it exactly as `gsettings` does, without this parsing. 

Tests would add clarity.

Reply:
Agreed that it's sorta spread out. I added some more tests for convertType, and added more comments to explain what getType does vs what convertType does.

Reply:
as far as returning exactly what gsettings outputs.. let's just say the gsettings cli tool uses very spartan type descriptions that I don't think are useful without the GVariant docs handy.